common/src/main/java/com/wynntils/screens/maps/SeaskipperMapScreen.java

```diff
@@ -0,0 +1,522 @@
+import com.wynntils.core.components.Managers;
+import com.wynntils.features.map.GuildMapFeature;
+import com.wynntils.models.items.items.gui.SeaskipperDestinationItem;
+import com.wynntils.models.map.pois.Poi;
+import com.wynntils.models.map.pois.SeaskipperPoi;
+import com.wynntils.screens.base.widgets.BasicTexturedButton;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.MapRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import com.wynntils.utils.type.BoundingBox;
+import com.wynntils.utils.wynn.ContainerUtils;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.components.events.GuiEventListener;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.phys.Vec3;
+import org.lwjgl.glfw.GLFW;
+
+public class SeaskipperMapScreen extends AbstractMapScreen {
+    private final AbstractContainerScreen<?> actualSeaskipperScreen;
+    private boolean hideTerritoryBorders = false;
+    private boolean renderAllDestinations = false;
+    private boolean renderRoutes = false;
+    private int boatSlot;
+    private final List<ItemStack> items = new ArrayList<>();
+    private final Map<SeaskipperDestinationItem, Integer> destinations = new HashMap<>();
+    private final List<Poi> seaskipperPois = new ArrayList<>();
+    private List<Poi> availableDestinations = new ArrayList<>();
+    private Poi currentPoi;
+
+    private SeaskipperMapScreen() {
+        if (McUtils.mc().screen instanceof AbstractContainerScreen<?> abstractContainerScreen) {
+            actualSeaskipperScreen = abstractContainerScreen;
+        } else {
+            throw new IllegalStateException(
+                    "Tried to open custom seaskipper screen when normal seaskipper screen is not open");
+        }
+
+        generateSeaskipperPois();
+    }
+
+    public static Screen create() {
+        return new SeaskipperMapScreen();
+    }
+
+    @Override
+    public void onClose() {
+        ContainerUtils.closeContainer(actualSeaskipperScreen.getMenu().containerId);
+        super.onClose();
+    }
+
+    public void addDestination(SeaskipperDestinationItem destination, int slot) {
+        destinations.put(destination, slot);
+    }
+
+    public void setBoatSlot(int slot) {
+        boatSlot = slot;
+    }
+
+    public void addItemStack(ItemStack item) {
```

Comment:
I think you can get away with using `abstractContainerScreen.getMenu().getItems()` so you don't have to collect items.